### ISSUE TITLE
Datastore.persistence is undefined.

### ISSUE THREAD

**Issue by apboon** [bajankristof/nedb-promises#4]
Datastore.persistence is undefined, therefore can't do NeDB documented functions like Datastore.persistence.compactDatafile();

Temporary workaround Datastore.__original.persistence.compactDatafile(); etc.

Reported this problem to both "nedb-promise" and "nedb-promises" projects. Similar project, similar problem :)

**Comment by bajankristof** [comment]
Hey there!
Sorry for the delayed answer, I have been busy.
It's been a long time and by now it is probably irrelevant but this should be fixed with 1.1.2's new `Datastore.create` method. If you want to, check out that version and see if this issue can be closed or not.

**Comment by dkebler** [comment]
This is still an issue 4.0.  Couldn't either access after create with `.persistence` or didn't grok how except via `.__original.persistence`

So decided to extend your class for easier access

Maybe some/all worthy of incorporating?

```javascript
class DataStore extends NEDBP {
  constructor(opts){
    super(opts)
  }

  indexes () {
    return this.__original.indexes
  }

  compact () {
    this.__original.persistence.compactDatafile()
  }

  autoCompact(interval) {
    this.__original.persistence.setAutocompactionInterval(interval)
  }

  async get (id) {
    let doc = await this.findOne({_id:id})
    return doc[id] ? doc[id] : doc
  }

  nedb() {
    return this.__original
  }
}
```

**Comment by catmade** [comment]
Same issue in 4.1.0 with TypeScript.

**Comment by catmade** [comment]
I use the following code to work correctly
```typescript
import Nedb from "nedb-promises";

let nedb = new Nedb({...});

let nedbAny: any = nedb;
nedbAny.__original.persistence.setAutocompactionInterval(1);
```

**Comment by bajankristof** [comment]
Can you elaborate what is and what isn't working? Is it the autocompletion? Does typescript not compile because it thinks `persistence` is not a property of the datastore? What's the issue here exactly?
Because in pure node JS using `Datastore.create` I can access `persistence` without any problem:


https://user-images.githubusercontent.com/15912759/103620726-4fc2eb00-4f34-11eb-9063-2c245bded71d.mov

**Comment by catmade** [comment]
It's just the **autocompletion** problem 😑! The [index.d.ts](https://github.com/bajankristof/nedb-promises/blob/master/index.d.ts) file caused this problem.

**Comment by bajankristof** [comment]
Thanks to @catmade the typescript definition file was fixed in #36 and published to npm in `4.1.1`. This should not be an issue from now on.
